jwcrypto/jws.py

```diff
@@ -264,7 +264,12 @@ def _verify(self, alg, key, payload, signature, protected, header=None):
                 raise InvalidJWSSignature('Invalid Unprotected header')
 
         # Merge and check (critical) headers
-        self._merge_check_headers(p, header)
+        chk_hdrs = self._merge_check_headers(p, header)
+        for hdr in chk_hdrs:
```

Comment:
I would expect headers checks to happen inside the function named _merge_check_headers() ?

Reply:
yea sounds good, will try that.

Reply:
This doesn't work well when I tried it. What ends up happening is that like in the header extension check function that i created in the test, which uses `jose_header`, it calls `self._merge_check_headers`, which ends up in a recursive loops.

Reply:
merge and check used to be two separate functions,
let's leave the code as you have it now with a FIXME comment, and I'll take care of restructuring at a later time.